tdmpc2/tdmpc2.py

```diff
@@ -218,48 +238,53 @@ def _td_target(self, next_z, reward, task):
 	def update(self, buffer):
 		"""
 		Main update function. Corresponds to one iteration of model learning.
-		
+
 		Args:
 			buffer (common.buffer.Buffer): Replay buffer.
-		
+
 		Returns:
 			dict: Dictionary of training statistics.
 		"""
 		obs, action, reward, task = buffer.sample()
-	
+		kwargs = {}
+		if task is not None:
+			kwargs["task"] = task
+		return self._update(obs, action, reward, **kwargs)
```

Comment:
we can't compile / record buffers yet :/
I'm working on this but there are some loose ends that I'm struggling to solve (recompile at each extension for instance), so I'm taking it away from the compiled code